python/triton/experimental/gluon/language/_semantic.py

```diff
@@ -1,142 +1,142 @@
-    return ttgl.shared_memory_descriptor(handle, **ty.__dict__)
-
-
-def memdesc_reshape(mem_desc, shape, layout, builder: GluonOpBuilder):
-    ty = ttgl.shared_memory_descriptor_type(mem_desc.dtype, shape, layout, mem_desc.type.alloc_shape)
-    handle = builder.create_memdesc_reshape(ty.to_ir(builder), mem_desc.handle)
-    return ttgl.shared_memory_descriptor(handle, **ty.__dict__)
-
-
-def memdesc_reinterpret(mem_desc, dtype, shape, layout, builder: GluonOpBuilder):
-    ty = ttgl.shared_memory_descriptor_type(dtype, shape, layout, shape)
-    handle = builder.create_memdesc_reinterpret(ty.to_ir(builder), mem_desc.handle)
-    return ttgl.shared_memory_descriptor(handle, **ty.__dict__)
-
-
-def warp_specialize(args, default_partition, worker_partitions, worker_num_warps: Sequence[int],
-                    worker_num_regs: Sequence[int], builder: GluonOpBuilder, generator):
-    num_partitions = len(worker_partitions)
-    assert num_partitions == len(
-        worker_num_warps), f"warp specialize got {num_partitions} partitions but {len(worker_num_warps)} warp counts"
-    assert num_partitions == len(
-        worker_num_regs), f"warp specialize got {num_partitions} partitions but {len(worker_num_regs)} register counts"
-
-    insert_pt = builder.get_insertion_point()
-
-    # Emit the default partition to get the result types.
-    default_block = builder.new_block()
-    builder.set_insertion_point_to_start(default_block)
-    default_results = generator.call_JitFunction(default_partition, args, kwargs={})
-    mlir_results = flatten_values_to_ir(default_results)
-    builder.create_warp_yield(mlir_results)
-    result_types = [r.get_type() for r in mlir_results]
-
-    # Create the warp specialize op.
-    builder.restore_insertion_point(insert_pt)
-    mlir_args = flatten_values_to_ir(args)
-    ws_op = builder.create_warp_specialize(result_types, mlir_args, worker_num_warps)
-    ws_op.get_default_region().push_back(default_block)
-    ws_op.set_requested_registers(worker_num_regs)
-
-    # Emit the partition regions.
-    builder.create_block_with_parent(ws_op.get_partition_op_holder(), [])
-    partitions_op = builder.create_warp_specialize_partitions(num_partitions)
-    arg_types = [arg.get_type() for arg in mlir_args]
-    for i in range(num_partitions):
-        block = builder.create_block_with_parent(partitions_op.get_region(i), arg_types)
-        block_args = [block.get_argument(j) for j in range(len(mlir_args))]
-        block_args = unflatten_ir_values(block_args, [arg.type for arg in args])
-        generator.call_JitFunction(worker_partitions[i], block_args, kwargs={})
-        builder.create_warp_return()
-
-    builder.set_insertion_point_after(ws_op.get_operation())
-    mlir_results = [ws_op.get_result(i) for i in range(len(result_types))]
-    return tuple(unflatten_ir_values(mlir_results, [r.type for r in default_results]))
+class GluonSemantic(TritonSemantic[ttgl.tensor]):
+    tensor = ttgl.tensor
```

Comment:
So Gluon and Triton still share the same tensor type but the tensor methods now dispatch to the language-specific semantic implementations?